specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/appconfiguration.json

```diff
@@ -1333,7 +1211,44 @@
       "type": "string",
       "x-ms-skip-url-encoding": true,
       "x-ms-parameter-location": "client"
-    }
+    },
+    "SelectSettings": {
+      "name": "$Select",
+      "in": "query",
+      "description": "Used to select what fields are present in the returned resource(s).",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": [
+          "key",
+          "label",
+          "content_type",
+          "value",
+          "last_modified",
+          "tags",
+          "locked",
+          "etag"
+        ],
+        "x-ms-client-name": "SettingFields"
```

Comment:
Also, can somebody please explain why this solution is preferable to `x-ms-enum`?

Reply:
I prefer the `x-ms-enum`.  However adding that is reported as a breaking change violation in the pipeline.  

cc: @JeffreyRichter , @mikekistler

Reply:
This is also a breaking change, so I'd like to understand the difference.

Reply:
@JeffreyRichter @mikekistler , Can you comment on Joel's question

Reply:
I don't understand what is happening here. Rick, did you make this PR/swagger change? Either change (this or x-ms-enum) is breaking as renaming the enum type is an SDK language is breaking. The linter tools should prevent unnamed enums to prevent more of this from happening. To fix these, we have to break. It's already broken since the enum type names may change anyway from code-gen to code-gen. We could force a 1-time break (in all language SDKs) to fix a swagger once and for all?

Reply:
Exactly, they're both breaking changes.  If the former isn't being flagged as such in CI then it's a bug in the CI tooling.

Reply:
Are we ok forcing the one time break with x-ms-enum to ensure codegen works correctly?  FWIW all SDKs are currently wrapping with HLC so it shouldn't actually break anyone

Reply:
Yes, I'll approve the break after you make it.